core_backend/app/db/db_models.py

```diff
@@ -316,124 +321,417 @@ class ContentDB(Base):
-        content_text=content.content_text,
-        content_language=content.content_language,
-        content_metadata=content.content_metadata,
+        content_title=content_text.content_title,
+        content_text=content_text.content_text,
+        content_id=content_text.content_id,
+        language=language,
+        content_metadata=content_text.content_metadata,
         updated_datetime_utc=datetime.utcnow(),
     )
 
     content_db = await asession.merge(content_db)
+    if content_db.content_id != old_content.content_id:
+        old_contents = await get_all_languages_version_of_content(
+            old_content.content_id, asession
+        )
+        if len(old_contents) < 1:
+            stmt = delete(ContentDB).where(
+                ContentDB.content_id == old_content.content_id
+            )
+            await asession.execute(stmt)
+
     await asession.commit()
     return content_db
 
 
-async def delete_content_from_db(
+async def delete_content_text_from_db(
+    content_text_id: int,
     content_id: int,
     asession: AsyncSession,
 ) -> None:
     """
-    Deletes a content from the database
+    Deletes a content text from the database
     """
-    stmt = delete(ContentDB).where(ContentDB.content_id == content_id)
+    stmt = delete(ContentTextDB).where(ContentTextDB.content_text_id == content_text_id)
     await asession.execute(stmt)
+
+    content_stmt = select(ContentTextDB).where(ContentTextDB.content_id == content_id)
+    content_row = (await asession.execute(content_stmt)).first()
+    if not content_row:
+        stmt = delete(ContentDB).where(ContentDB.content_id == content_id)
+        await asession.execute(stmt)
     await asession.commit()
 
 
 async def get_content_from_db(
-    content_id: int,
+    content_text_id: int,
     asession: AsyncSession,
-) -> Optional[ContentDB]:
+) -> Optional[ContentTextDB]:
     """
     Retrieves a content from the database
     """
-    stmt = select(ContentDB).where(ContentDB.content_id == content_id)
-    content_row = (await asession.execute(stmt)).first()
-    if content_row:
-        return content_row[0]
-    else:
-        return None
+    stmt = select(ContentTextDB).where(ContentTextDB.content_text_id == content_text_id)
+    content_row = (await asession.execute(stmt)).scalar_one_or_none()
+    return content_row
 
 
 async def get_list_of_content_from_db(
     asession: AsyncSession, offset: int = 0, limit: Optional[int] = None
-) -> List[ContentDB]:
+) -> List[ContentTextDB]:
+    """
+    Retrieves all contents from the database
+    """
+    stmt = select(ContentTextDB)
+    if offset > 0:
+        stmt = stmt.offset(offset)
+    if limit is not None:
+        stmt = stmt.limit(limit)
+    content_rows = (await asession.execute(stmt)).all()
+
+    return [c[0] for c in content_rows] if content_rows else []
+
+
+async def get_summary_of_content_from_db(
+    asession: AsyncSession,
+    language_id: int,
```

Comment:
Right now I can  query /summary endpoint with an nonexistent language ID and would get an empty list. We should check if the language ID exists, and if not raise an error. For languages that do exist, returning empty list makes sense!

Reply:
Btw feel free to turn some of these into backlog tickets!